Simplenote/src/test/java/com/automattic/simplenote/viewmodels/TagsViewModelTest.kt

```diff
@@ -0,0 +1,232 @@
+        assertNull(viewModel.uiState.value?.searchQuery)
+    }
+
+    @Test
+    fun searchShouldFilterTags() = runBlockingTest {
+        fakeTagsRepository.stub {
+            onBlocking { allTags() }.doReturn(tagItems)
+        }
+        fakeTagsRepository.stub {
+            onBlocking { tagsChanged() }.doReturn(emptyFlow())
+        }
+        viewModel.start()
+
+        val filteredList = listOf(tagItems[1])
+        val searchQuery = "tag2"
+        fakeTagsRepository.stub {
+            onBlocking { searchTags(any()) }.doReturn(filteredList)
+        }
+        viewModel.search(searchQuery)
+
+        // Flay tests regarding runBlockingTest. I will include this sleep here for now
+        // while we find the bug in runBlockingTest
+        Thread.sleep(500)
+
+        assertEquals(viewModel.uiState.value?.tagItems, filteredList)
+        assertEquals(viewModel.uiState.value?.searchUpdate, true)
+        assertEquals(viewModel.uiState.value?.searchQuery, "tag2")
+    }
+
+    @Test
+    fun afterAddingATagUpdateOnResultShouldUpdateUiState() = runBlockingTest {
+        fakeTagsRepository.stub {
+            onBlocking { allTags() }.doReturn(tagItems)
+        }
+        fakeTagsRepository.stub {
+            onBlocking { tagsChanged() }.doReturn(emptyFlow())
+        }
+        viewModel.start()
+
+        // Add a new tag
+        tagItems.add(TagItem(Tag("tag10"), 2))
+
+        fakeTagsRepository.stub {
+            onBlocking { allTags() }.doReturn(tagItems)
+        }
+
+        viewModel.updateOnResult()
+
+        assertEquals(viewModel.uiState.value?.tagItems, tagItems)
+        assertEquals(viewModel.uiState.value?.searchUpdate, false)
+        assertNull(viewModel.uiState.value?.searchQuery)
+    }
+
+    @Test
+    fun clickEditTagShouldTriggerEventEditTagEvent() {
+
+        viewModel.clickEditTag(tagItems[0])
+
+        assertEquals(viewModel.event.value, TagsEvent.EditTagEvent(tagItems[0]))
+    }
+
+    @Test
+    fun clickDeleteTagOfTagWithNotesShouldTriggerDeleteTagEvent() {
+        viewModel.clickDeleteTag(tagItems[1])
+
+        assertEquals(viewModel.event.value, TagsEvent.DeleteTagEvent(tagItems[1]))
+    }
+
+    @Test
+    fun clickDeleteTagOfTagWithNotesShouldDeleteTagDirectly() = runBlockingTest {
+        fakeTagsRepository.stub {
+            onBlocking { deleteTag(any()) }.doReturn(Unit)
+        }
+
+        fakeTagsRepository.stub {
+            onBlocking { tagsChanged() }.doReturn(flow{ emit(true) })
+        }
+
+        val updatedTags = tagItems.filter { tagItem -> tagItem.tag.name != "tag2" }
+        fakeTagsRepository.stub {
+            onBlocking { allTags() }.doReturn(updatedTags)
+        }
+
+        viewModel.clickDeleteTag(tagItems[1])
+
+        fakeTagsRepository.stub {
+            onBlocking { allTags() }.doReturn(updatedTags)
+        }
+        viewModel.updateOnResult()
+
+        // Flay tests regarding runBlockingTest. I will include this sleep here for now
```

Comment:
typo: Flay -> Flaky?

Reply:
Yes, typo. But we already removed the sleep call since we fixed the tests in 87b060c762f390a5421c6821e8c7dc96751c2621.